pandas/core/internals/blocks.py

```diff
@@ -2436,6 +2465,12 @@ def _can_hold_element(self, element):
     def _try_coerce_args(self, values, other):
         """ provide coercion to our input arguments """
 
+        if isinstance(other, DatetimeArray):
```

Comment:
seems maybe can combine with the code below with a common class for DTA and DTI?